packages/server/src/games.ts

```diff
@@ -58,3 +58,99 @@ export function updateGameState(io: IOServer, store: Store, roomName: string, ga
     io.in(roomName).emit(GameEvent.CurrentState, updatedGameState);
   }
 }
+
+/**
+ * Calculate the immediate new game state, as a result of a player's action. This is not necessarily the final state
+ * but an intermediate one to help the following calculations.
+ */
+function calculatePlayerAction(oldState: GameState, playerAction: PlayerAction) {
+  const activePlayer = oldState.players.filter((playerState) => playerState.username === playerAction.playerName)[0];
```

Comment:
you can destructure the array and do `const [activePlayer] = ...` instead of doing `const a = array[0]` but just a syntax thing, up to you if you prefer this way

Reply:
Ah yeah! Thanks, I was actually googling that and didn't see it.